lighthouse-core/gather/driver.js

```diff
@@ -418,17 +418,89 @@ class Driver {
     });
   }
 
+  /**
+   * Deprecated: renamed to `evaluate`.
+   * @param {string} expression
+   * @param {{useIsolation?: boolean}=} options
+   * @return {Promise<*>}
+   */
+  async evaluateAsync(expression, options = {}) {
+    return this.evaluate(expression, options);
+  }
+
+  /**
+   * Evaluate an expression in the context of the current page. If useIsolation is true, the expression
+   * will be evaluated in a content script that has access to the page's DOM but whose JavaScript state
+   * is completely separate.
+   * Returns a promise that resolves on the expression's value.
+   * See the documentation for `createEvalCode` in `page-functions.js`.
+   * @template T, R
+   * @param {string | ((...args: T[]) => R)} expressionOrMainFn
+   * @param {{useIsolation?: boolean, args?: T[], deps?: Function[]}=} options
+   * @return {Promise<R>}
+   */
+  async evaluate(expressionOrMainFn, options = {}) {
+    if (typeof expressionOrMainFn !== 'string') {
+      expressionOrMainFn = pageFunctions.createEvalCode(expressionOrMainFn, {
+        mode: 'iffe',
+        ...options,
+      });
+    }
+    return this._evaluate(expressionOrMainFn, options);
+  }
+
+  /**
+   * Evaluate a statement in the context of a JavaScript object on the current page.
+   * Returns a promise that resolves on the expression's value.
+   * See the documentation for `createEvalCode` in `page-functions.js`.
+   * @template T, R
+   * @param {string | ((...args: T[]) => R)} statementOrMainFn
+   * @param {{objectId: string, args?: T[], deps?: Function[]}} options
+   * @return {Promise<R>}
+   */
+  async evaluateFunctionOnObject(statementOrMainFn, options) {
```

Comment:
possible to split this into a separate PR since it's mostly an independent change?

Reply:
will do